input/CatalogLoading.cpp

```diff
@@ -220,8 +220,14 @@ RegionCatalog loadLocusCatalogFromDisk(
     for (auto& locusJson : catalogJson)
     {
         LocusDescriptionFromUser userDescription = loadUserDescription(locusJson, reference.contigInfo());
-        LocusSpecification locusSpec = decodeLocusSpecification(userDescription, sampleSex, reference, heuristicParams);
-        catalog.emplace(std::make_pair(locusSpec.locusId(), locusSpec));
+        try {
+	    LocusSpecification locusSpec = decodeLocusSpecification(userDescription, sampleSex, reference, heuristicParams);
+	    catalog.emplace(std::make_pair(locusSpec.locusId(), locusSpec));
+        }
+	catch (string s) {
```

Comment:
How about replacing the catch statement with something like below?

```c++
catch (const std::exception& except) {
  const string message = "Unable to load " + locusSpec.locusId() + ": " + except.what();

  if (heuristicParams.treatWarningsAsErrors())
  {
    throw std::runtime_message(message);
  }
  else
  {
    console->warn(message);
  }
}
```

Reply:
Sounds great! A couple questions:

1) To make locusSpec in scope in the catch statement, I changed:

```
try {
    LocusSpecification locusSpec = decodeLocusSpecification(userDescription, sampleSex, reference, heuristicParams);
    catalog.emplace(std::make_pair(locusSpec.locusId(), locusSpec));
}
```

to 

```
LocusSpecification locusSpec;
try {
    locusSpec = decodeLocusSpecification(userDescription, sampleSex, reference, heuristicParams);
    catalog.emplace(std::make_pair(locusSpec.locusId(), locusSpec));
}
```

But then I get the error message:

```
/hpf/largeprojects/tcagstor/users/btrost/EH_pull2/ExpansionHunter/input/CatalogLoading.cpp: In function ‘ehunter::RegionCatalog ehunter::loadLocusCatalogFromDisk(const string&, ehunter::Sex, const ehunter::HeuristicParameters&, const ehunter::Reference&)’:
/hpf/largeprojects/tcagstor/users/btrost/EH_pull2/ExpansionHunter/input/CatalogLoading.cpp:223:28: error: no matching function for call to ‘ehunter::LocusSpecification::LocusSpecification()’
         LocusSpecification locusSpec;
```

Not sure why...? Isn't `LocusSpecification locusSpec;` just declaring a variable call locusSpec of type LocusSpecification?

2) Where does std::runtime_message come from? When I try to compile, I get a message saying it is not defined.

Reply:
1. My fault -- there is a better way to do this. `LocusSpecification locusSpec;` does not work because there is no 
  way to create a variable/object of type/class `LocusSpecification` without providing additional parameters 
  required by the constructor (code that sets up the variable).
  How about something like `const string message = "Unable to load " + locusJson.dump() + ": " + except.what();`? This way we can isolate `locusSpec` in the try block.
2. This is supposed to be `std::runtime_error` from `<stdexcept>`